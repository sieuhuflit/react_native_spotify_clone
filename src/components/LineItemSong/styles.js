import { StyleSheet } from 'react-native';
import { colors } from '../../constants';

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    flexDirection: 'row',
    justifyContent: 'space-between',
    padding: 16,
    width: '100%',
  },
  title: {
    fontSize: 16,
    color: colors.white,
    marginBottom: 4,
  },
  circleDownloaded: {
    alignItems: 'center',
    backgroundColor: colors.brandPrimary,
    borderRadius: 7,
    height: 14,
    justifyContent: 'center',
    marginRight: 8,
    width: 14,
  },
  artist: {
    fontSize: 12,
    color: colors.greyInactive,
  },
  containerRight: {
    alignItems: 'flex-end',
    flex: 1,
  },
});

export default styles;
