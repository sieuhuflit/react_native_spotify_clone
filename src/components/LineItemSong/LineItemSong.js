import React from 'react';
import PropTypes from 'prop-types';
import { Text, TouchableOpacity, View } from 'react-native';
import { get } from 'lodash';
import Feather from 'react-native-vector-icons/Feather';
import { colors, globalStyles } from '../../constants';
import styles from './styles';

const LineItemSong = ({ active, onPress, data }) => {
  const songName = get(data, 'name', '');
  const artistName = get(data, 'artists[0].name', '');
  return (
    <View style={styles.container}>
      <TouchableOpacity
        activeOpacity={globalStyles.activeOpacity}
        onPress={() => onPress(data)}
        style={globalStyles.flex5}
      >
        <Text style={[styles.title, { color: active ? colors.brandPrimary : colors.white }]}>
          {songName}
        </Text>
        <View style={globalStyles.flexRow}>
          <Text style={styles.artist}>{artistName}</Text>
        </View>
      </TouchableOpacity>
      <View style={styles.containerRight}>
        <Feather color={colors.greyLight} name="more-horizontal" size={20} />
      </View>
    </View>
  );
};

LineItemSong.propTypes = {
  onPress: PropTypes.func.isRequired,
  data: PropTypes.shape({
    name: PropTypes.string,
    artists: PropTypes.array,
  }),
  active: PropTypes.bool,
};

LineItemSong.defaultProps = {
  active: false,
  data: {
    name: '',
    artists: [],
  },
};

export default LineItemSong;
