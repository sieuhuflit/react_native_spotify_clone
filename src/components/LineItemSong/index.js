import LineItemSong from './LineItemSong';

export default LineItemSong;
