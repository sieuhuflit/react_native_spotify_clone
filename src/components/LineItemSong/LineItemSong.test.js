import React from 'react';
import { shallow } from 'enzyme';
import LineItemSong from './LineItemSong';

const setup = propOverrides => {
  const props = Object.assign(
    {
      onPress: jest.fn()
    },
    propOverrides
  );
  const wrapper = shallow(<LineItemSong {...props} />);
  return {
    props,
    wrapper
  };
};

describe('LineItemSong check snapshot', () => {
  test('should match snapshot', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });
});

describe('LineItemSong check render correctly', () => {
  test('should render correctly when data is null ', () => {
    const { wrapper } = setup({
      data: {
        name: 'Test'
      }
    });
    expect(
      wrapper
        .find('Text')
        .children()
        .at(0)
        .text()
    ).toEqual('Test');
  });
});

describe('LineItemSong check simulate click', () => {
  test('should onPress work', () => {
    const { wrapper, props } = setup();
    wrapper
      .find('TouchableOpacity')
      .props()
      .onPress();
    expect(props.onPress).toBeCalledTimes(1);
  });
});
