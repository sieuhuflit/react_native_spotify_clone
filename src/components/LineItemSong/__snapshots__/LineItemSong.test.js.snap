// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LineItemSong check snapshot should match snapshot 1`] = `ShallowWrapper {}`;
